crates/project/src/terminals.rs

```diff
@@ -277,6 +280,20 @@ impl Project {
         settings: &TerminalSettings,
         cx: &AppContext,
     ) -> Option<PathBuf> {
+        if let Some((worktree, _)) = self.find_worktree(abs_path, cx) {
+            let toolchain = self
+                .active_toolchain(worktree.read(cx).id(), LanguageName::new("Python"), cx)
+                .now_or_never()
```

Comment:
We should probably not do `now_or_never` here, since `active_toolchain` might roundtrip to a remote SSH server; I'd suggest turning this function into one that returns `Task<Option<PathBuf>>`.

Reply:
Alternatively, you could do a `block_on` for this task; this is slightly better than now_or_never, as it will ensure that the remote case is handled well.

Reply:
Actually, it might be that this function only ever runs in local environments, in which we don't need to give a dime about remote use case; in such cases now_or_never should be ok.

Reply:
How are you sure this is only being executed locally? If not, I can also change it to use `block_on` - can you show me an example here? (somehow I find multiple packages providing this function)

Reply:
> How are you sure this is only being executed locally?

I've ran your branch using ssh onto localhost; it wouldn't work. Sadly, I tried `smol::block_on` and than doesn't work either, which means that we'll have to turn this function into an async one. Yikes.. :)
I'll push onto your branch if you don't mind

Reply:
Appreciate the time you take to explain things and help me out! Definitely makes me want to pick up Rust and help out more with the development of Zed :)

Yes, feel free to push to this branch.

Reply:
Ok, 2368b2132118e7a49c3e306a66f68ed0345d9656 makes it work with ssh, with a small caveat that there's a bug; we don't have an active toolchain before user opens up any Python file. I think we can fix that within workspace deserialization code; there will still be an edge case that the terminal won't have venv activated up until you pick a venv for the first time, but the rest should work fine-ish.

Reply:
Nvm lol, I just didn't have a venv picked in my test project. It works like a charm. :)